test_fatoptimizer.py

```diff
@@ -2829,6 +2829,19 @@ def f(x):
                 return 42 + 3
         ''')
 
+    def test_nested_function(self):
+        self.check_optimize('''
+            def f(x):
+                def g(x):
+                    return 100
+                return g(x) + 3
```

Comment:
Guard on the inline functions:
http://fatoptimizer.readthedocs.org/en/latest/fat.html#GuardFunc

Oh by the way, currently the guard only watch for func.__code__. For function inlining, it's probably not enough, we need also to ensure that default values of parameters didn't change, and other sublte properties of a function.

I didn't have time yet to investigate this. To implement an efficient guard, we may have to also add a "version tag" to functions, not only to dicts (PEP 509). Types already have a version tag since Python 2.6.